op-service/dial/active_l2_provider.go

```diff
@@ -79,8 +93,8 @@ func (p *ActiveL2EndpointProvider) EthClient(ctx context.Context) (EthClientInte
 	if err != nil {
 		return nil, err
 	}
-	if p.ethClientIndex != p.rollupIndex || p.currentEthClient == nil {
```

Comment:
I see that the logic now got mostly duplicated into the ctor, but couldn't be removed here. As it was before was then probably better after all. Sorry didn't realize it would turn out like this.

Reply:
No worries, I like it either way. I'll go back to the old way.